utils/utils_quantization.py

```diff
@@ -0,0 +1,312 @@
+    ax.annotate('$\partial \Omega$', [-1.3, -1.9], fontsize=24)
+    #ax.annotate('$N(\mathbf{c})$', [-6.5,0], fontsize=24, rotation=45)
+
+    ax.set_axis_off()
+    ax.set_ylim(ymin,ymax)
+    ax.set_xlim(xmin, xmax)
+
+
+def plot_dgm(dgm, box=None, ax=None, color="blue", label='diagram', alpha=None):
+    if ax is None:
+        fig = plt.figure()
+        ax = fig.add_subplot(111)
+    if dgm.size:
+        x, y = dgm[:, 0], dgm[:, 1]
+        ax.scatter(x, y, marker='o', color=color, label=label, alpha=alpha)
+    if box is None:
+        if dgm.size:
+            m, M = np.min(dgm) - 0.1, np.max(dgm) + 0.1
+        else:
+            m, M = 0, 1
+    else:
+        m, M = box
+    ax.set_xlim(m, M)
+    ax.set_ylim(m, M)
+    ax.add_patch(mpatches.Polygon([[m, m], [M, m], [M, M]], fill=True, color='lightgrey'))
+    ax.set_aspect("equal")
+    ax.set_xlabel("Birth", fontsize=24)
+    ax.set_ylabel("Death", fontsize=24)
+    # ax.legend()
+
+
+########################
+### Experiment utils ###
+########################
+
+
+def _sample_torus(n, r1, r2, radius_eps, ambiant_eps):
+    # Sample points uniformly on a torus of big radius r1 and small radius r2
+    theta1 = 2 * np.pi * np.random.rand(n)
+    theta2 = 2 * np.pi * np.random.rand(n)
+
+    r1 = r1 + radius_eps * (2 * np.random.rand() - 1)
+    r2 = r2 + radius_eps * (2 * np.random.rand() - 1)
+
+    x = (r1 + r2 * np.cos(theta2)) * np.cos(theta1)
+    y = (r1 + r2 * np.cos(theta2)) * np.sin(theta1)
+    z = r2 * np.sin(theta2)
+
+    X = np.array([x, y, z]).T
+
+    X = X + ambiant_eps * (2 * np.random.rand(n,3) - 1)
+
+    return X
+
+
+def _compute_pd(X, hdim=1, min_persistence=0.0001, mode="alpha", rotate=False):
+    if mode == "alpha":
+        ac = gd.AlphaComplex(points=X)
+        st = ac.create_simplex_tree()
+    elif mode == "rips":
+        ac = gd.RipsComplex(points=X)
+        st = ac.create_simplex_tree(max_dimension=2)
+    pers = st.persistence(min_persistence=min_persistence)
+    h1 = st.persistence_intervals_in_dimension(hdim)
+    if mode == "alpha":
+        h1 = np.sqrt(np.array(h1))
+        if rotate:
+            h1[:, 1] = h1[:, 1] - h1[:, 0]
+        return h1
+    else:
+        return np.array(h1) / 2  # to make it comparable with Cech
+
+
+def build_dataset(K, params):
+    average_nb_pts = params['nb_points']
+    ns = np.random.poisson(lam=average_nb_pts, size=K)
+    r1, r2 = params['r1'], params['r2']
+    radius_eps = params['radius_eps']
+    ambiant_eps = params['ambiant_eps']
+    Xs = [_sample_torus(n, r1, r2, radius_eps, ambiant_eps) for n in ns]
+    diags = [_compute_pd(X) for X in Xs]
+    return Xs, diags
+
+
+##############################
+### Quantization algorithm ###
+##############################
+
+
+
+
+def _dist_to_diag(X, internal_p):
+    return ((X[:, 1] - X[:, 0]) * 2 ** (1. / internal_p - 1))
+
+
+def _build_dist_matrix(X, Y, order=2., internal_p=2):
```

Comment:
As we were discussing, maybe this function could become public at some point.

Reply:
Yes